### ISSUE TITLE
constant JSON::Parser not defined (NameError)

### ISSUE THREAD

**Issue by shaicoleman** [flori/json#287]
After requiring the json_pure 2.0, it breaks rubocop,

To replicate the issue:
Clone https://github.com/bbatsov/rubocop
Add to Gemfile: gem 'json_pure', '~> 2.0.0'  and run bundle install
run rspec

**Comment by shaicoleman** [comment]
```
gems/json_pure-2.0.0/lib/json/common.rb:28:in `remove_const': constant JSON::Parser not defined (NameError)
  from gems/json_pure-2.0.0/lib/json/common.rb:28:in `parser='
  from gems/json_pure-2.0.0/lib/json/ext.rb:10:in `<module:Ext>'
  from gems/json_pure-2.0.0/lib/json/ext.rb:6:in `<module:JSON>'
  from gems/json_pure-2.0.0/lib/json/ext.rb:3:in `<top (required)>'
  from gems/json_pure-2.0.0/lib/json.rb:59:in `require'
  from gems/json_pure-2.0.0/lib/json.rb:59:in `<module:JSON>'
  from gems/json_pure-2.0.0/lib/json.rb:55:in `<top (required)>'
  from rubocop/lib/rubocop/formatter/json_formatter.rb:4:in `require'
  from rubocop/lib/rubocop/formatter/json_formatter.rb:4:in `<top (required)>'
  from rubocop/lib/rubocop.rb:401:in `require'
  from rubocop/lib/rubocop.rb:401:in `<top (required)>'
  from rubocop/spec/spec_helper.rb:11:in `require'
  from rubocop/spec/spec_helper.rb:11:in `<top (required)>'
  from rubocop/spec/isolated_environment_spec.rb:4:in `require'
  from rubocop/spec/isolated_environment_spec.rb:4:in `<top (required)>'
  from gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1361:in `load'
  from gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1361:in `block in load_spec_files'
  from gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1359:in `each'
  from gems/rspec-core-3.4.4/lib/rspec/core/configuration.rb:1359:in `load_spec_files'
  from gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:106:in `setup'
  from gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:92:in `run'
  from gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:78:in `run'
  from gems/rspec-core-3.4.4/lib/rspec/core/runner.rb:45:in `invoke'
  from gems/rspec-core-3.4.4/exe/rspec:4:in `<top (required)>'
  from bin/rspec:23:in `load'
  from bin/rspec:23:in `<main>'
  from bin/ruby_executable_hooks:15:in `eval'
  from bin/ruby_executable_hooks:15:in `<main>'

```

**Comment by brugidou** [comment]
https://github.com/flori/json/commit/79074a0de768113364749cb584000b7e97be04e5#diff-1fb83c48ff17d82315a53e5f26bc4996R27

We should set the `const_defined?` mod to `false` to avoid checking the root namespace for `::Parser` (which is probably defined in rubocop)

**Comment by reggieb** [comment]
See my comments in #288 

The older version of the parse method looks to be less problematic.

**Comment by flori** [comment]
Can you try again with version 2.0.1?

**Comment by revett** [comment]
@flori - Can confirm that this is fixed in **v2.0.1** 🎉

**Comment by flori** [comment]
Great! ⇒ 🚴
